uninstall.sh

```diff
@@ -0,0 +1,292 @@
+  elif [ -f /etc/SuSe-release ]; then
+    # Older SuSE/etc.
+    OS="SuSE"
+    OS_VER="?"
+  elif [ -f /etc/redhat-release ]; then
+    # Older Red Hat, CentOS, etc.
+    OS="Red Hat/CentOS"
+    OS_VER="?"
+  else
+    # Fall back to uname, e.g. "Linux <version>", also works for BSD, etc.
+    OS=$(uname -s)
+    OS_VER=$(uname -r)
+  fi
+
+  OS=$(echo "$OS" | awk '{print tolower($0)}')
+  OS_VER_MAJOR=$(echo "$OS_VER" | cut -d. -f1)
+}
+
+check_os_comp() {
+  SUPPORTED=false
+  case "$OS" in
+  ubuntu)
+    [ "$OS_VER_MAJOR" == "18" ] && SUPPORTED=true
+    [ "$OS_VER_MAJOR" == "20" ] && SUPPORTED=true
+    ;;
+  debian)
+    [ "$OS_VER_MAJOR" == "9" ] && SUPPORTED=true
+    [ "$OS_VER_MAJOR" == "10" ] && SUPPORTED=true
+    ;;
+  centos)
+    [ "$OS_VER_MAJOR" == "7" ] && SUPPORTED=true
+    [ "$OS_VER_MAJOR" == "8" ] && SUPPORTED=true
+    ;;
+  esac
+
+  # exit if not supported
+  if [ "$SUPPORTED" == true ]; then
+    echo "* $OS $OS_VER is supported."
+  else
+    echo "* $OS $OS_VER is not supported"
+    error "Unsupported OS"
+    exit 1
+  fi
+}
+
+### Main uninstallation functions ###
+
+rm_panel_files() {
+  rm -rf /var/www/pterodactyl /usr/local/bin/composer
+  [ "$OS" != "centos" ] && unlink /etc/nginx/sites-enabled/pterodactyl.conf
+  [ "$OS" != "centos" ] && rm -f /etc/nginx/sites-available/pterodactyl.conf
+  [ "$OS" == "centos" ] && rm -f /etc/nginx/conf.d/pterodactyl.conf
+}
+
+rm_wings_files() {
+  rm -rf /etc/pterodactyl /usr/local/bin/wings /var/lib/pterodactyl
+}
+
+rm_services() {
+  systemctl disable --now mariadb
+  systemctl disable --now pteroq
+  rm -rf /etc/systemd/system/pteroq.service
+  case "$OS" in
+  debian | ubuntu)
+    systemctl disable --now redis-server
+    ;;
+  centos)
+    systemctl disable --now redis
+    systemctl disable --now php-fpm
+    rm -rf /etc/php-fpm.d/www-pterodactyl.conf
+    ;;
+  esac
+}
+
+rm_cron() {
+  # shellcheck disable=SC2063
+  crontab -l | grep -v "* * * * * php /var/www/pterodactyl/artisan schedule:run >> /dev/null 2>&1" | crontab -
+}
+
+rm_database() {
+  valid_db=$(mysql -u root -p -e "SELECT schema_name FROM information_schema.schemata;" | grep -v -E -- 'schema_name|information_schema|performance_schema')
+  warning "Be careful! This database will be deleted!"
+  if [[ "$valid_db" == *"panel"* ]]; then
+    echo -n "* Database called panel has been detected. Is it the pterodactyl database? (y/N): "
+    read -r is_panel
+    [[ "$is_panel" =~ [Yy] ]] && DATABASE=panel || echo "$valid_db"
+  else
+    echo "$valid_db"
+  fi
+  while [ -z "$DATABASE" ] || [[ $valid_db != *"$database_input"* ]]; do
+    echo -n "* Choose the panel database(to skip type none): "
+    read -r database_input
+    DATABASE=$database_input
+  done
+  [[ "$DATABASE" != "none" ]] && mysql -u root -p -e "DROP $DATABASE"
+  # Exclude usernames User and root (Hope no one uses username User)
+  valid_users=$(mysql -u root -p -e "SELECT user FROM mysql.user;" | grep -v -E -- 'User|root')
```

Comment:
Why are we excluding `User`? What's so specific about that user?

Reply:
When you use the `SELECT user FROM mysql.user;` it gives back a table with a column called user. So it needs to be excluded as it will be always when you run that command. But I just noticed, that I wrote it `User` not `user`.